From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Tue, 18 Feb 2020 20:27:15 +0100
Subject: Skip time-zone tests that fails with system ICU

There have been various changes upstream as part of the revision [1] that
change the handling of timezones to use ICU all the times when available,
and by "available" they meant only in the !MOZ_SYSTEM_ICU case, and
changing the system-ICU code path to fix it, is not trivial.

One of the bugs fixed with this change is this one (TZ dependent):

  >>> new Date(1970,0,1,0,0,0,0)
  Thu Jan 01 1970 01:00:00 GMT+0100 (IST) // Expected time 00:00:00 ...
  >>> new Date(1970,0,1,0,0,0,0).getHours()
  1 // Expected 0

Not being a regression, but an 11-years old (at the date), I think we can
sadly accept it, until we don't start compile SpiderMonkey to use Mozilla's
ICU.

[1] https://hg.mozilla.org/mozilla-central/rev/7b1a3a49547d

Forwarded: no
Bug-Mozilla: https://bugzilla.mozilla.org/show_bug.cgi?id=487897
Bug-Mozilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1346211
---
 js/src/tests/jstests.list | 6 ++++++
 1 file changed, 6 insertions(+)

Index: firefox-68.6.0/js/src/tests/jstests.list
===================================================================
--- firefox-68.6.0.orig/js/src/tests/jstests.list
+++ firefox-68.6.0/js/src/tests/jstests.list
@@ -503,6 +503,12 @@ skip script non262/Intl/DateTimeFormat/t
 skip script non262/Intl/NumberFormat/formatToParts.js
 skip script non262/Intl/RelativeTimeFormat/format.js
 
+# These tests fails when using system ICU because the related bug has been fixed only when using local ICU
+# See https://bugzilla.mozilla.org/show_bug.cgi?id=1346211#c7
+# https://hg.mozilla.org/mozilla-central/rev/7b1a3a49547d
+skip script non262/Date/time-zones-historic.js
+skip script non262/Date/time-zones-imported.js
+
 ###########################################################
 # Tests disabled due to issues in test262 importer script #
 ###########################################################
